### ISSUE TITLE
If statement in .rambaspec

### ISSUE THREAD

**Issue by Antowkos** [rambler-digital-solutions/Generamba#128]
I use different classes for HTTP request, it parameters, path components and headers and it would be great if i could use only one template and just set some custom parameter like `create_path_components:true` and then path components class creates. Because not every request need to have path components or some parameters.

**Comment by Beniamiiin** [comment]
Hi, @Antowkos. In last release we implement parameter `custom-parameters` for `generamba gen` command. Here is an example: `generamba gen my_module my_template --custom-parameters=key1:value1 key2:value2`. The value1 will be placed in your module for `{{ module_info.custom_parameters.key1 }}`. This is what you need?

**Comment by Antowkos** [comment]
Hi, @Beniamiiin. Thanks for so fast answer. Could i use it in `.rambaspec` file? I thought that `{{ module_info.custom_parameters.key1 }}` is only available in `.liquid` files for templates.

**Comment by etolstoy** [comment]
@Antowkos in what fields of `.rambaspec` do you want to use these parameters?

**Comment by Antowkos** [comment]
@etolstoy In `code_files` and `test_files`. But maybe it will be useful in `dependencies` too.

**Comment by etolstoy** [comment]
@Antowkos you can already use these parameters: https://github.com/rambler-ios/Generamba/pull/113

**Comment by Antowkos** [comment]
@etolstoy , @Beniamiiin . Correct me if i'm wrong. As i understand `.rambaspec` file is written in YAML and if it is then i don't know how i can use `if` statement there. Could you provide an example of how i can use `if` statement in `.rambaspec` file to add different files from one template base on `custom-parameters` key?

**Comment by Beniamiiin** [comment]
@Antowkos if i understood you correctly, you want to do something like?

```
{% if create_path_components == true %}
   - {name: View/MyCustomViewInput.h, path: Code/View/view_input.h.liquid}
{% else %}
   - {name: View/ViewInput.h, path: Code/View/view_input.h.liquid}
{% endif %}
```

**Comment by Antowkos** [comment]
@Beniamiiin Yes. Is it possible now?

**Comment by Beniamiiin** [comment]
@Antowkos unfortunately no, but we implement this ability in next release.

**Comment by Antowkos** [comment]
@Beniamiiin Cool! Thanks! 👍

**Comment by Antowkos** [comment]
@Beniamiiin Hi again! I just tried to use this new feature and it doesn't work.

I use syntax like this:

```
{% if create_path_components == true %}
   - {name: View/MyCustomViewInput.h, path: Code/View/view_input.h.liquid}
{% else %}
   - {name: View/ViewInput.h, path: Code/View/view_input.h.liquid}
{% endif %}
```

And got this error:

```
some_path/.rvm/rubies/ruby-2.2.4/lib/ruby/2.2.0/psych.rb:370:in `parse': (some_path/spec_name.rambaspec): found character that cannot start any token while scanning for the next token at line 11 column 52 (Psych::SyntaxError)
```

If i understand code of creation right then the error could be placed in creation of object `ModuleTemplate`. It waits for `options` to be passed to parse `.rambaspec` file as file with liquid markup, but it creates only in one place (`lib/generamba/cli/gen_command.rb`, line 48) this way `template = ModuleTemplate.new(template_name)`, there are no options passed, as you can see, so i think this is the cause.

**Comment by Antowkos** [comment]
@Beniamiiin @etolstoy Is anyone saw my comment?

**Comment by Beniamiiin** [comment]
@Antowkos hi. Unfortunately we lost couple of code lines, when merged :) I already fixed this bug and we will release new version today.

**Comment by Antowkos** [comment]
Cool, thanks!

**Comment by etolstoy** [comment]
Ready: https://github.com/rambler-ios/Generamba/releases/tag/1.2.1

**Comment by MaximBazarov** [comment]
seems like don't work, this is `.rambaspec`

``` ruby
# Template information section
name: nk-vicr-p
summary: NavigationKit based VICR module
author: Maksim Bazarov 
version: 1.0.0
license: MIT

# The declarations for code files
code_files:
- {name: Assembly.swift, path: Code/assembly.swift.liquid}
- {name: Module/View/View.swift, path: Code/view.swift.liquid}

{% if custom_parameters.type == 'table' %}
- {name: Module/View/DataSource.swift, path: Code/datasource.swift.liquid}
{% endif %}
```

And this is an error 

``` sh
ruby/2.3.0/psych.rb:377:in `parse': (swift-templates/nk-vicr-p/nk-vicr-p.rambaspec): found character that cannot start any token while scanning for the next token at line 13 column 2 (Psych::SyntaxError)
     lib/ruby/2.3.0/psych.rb:377:in `parse_stream'
     lib/ruby/2.3.0/psych.rb:325:in `parse'
     lib/ruby/2.3.0/psych.rb:252:in `load'
     lib/ruby/2.3.0/psych.rb:471:in `block in load_file'
     lib/ruby/2.3.0/psych.rb:471:in `open'
     lib/ruby/2.3.0/psych.rb:471:in `load_file'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/template/helpers/rambaspec_validator.rb:25:in `validate_spec'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/template/installer/local_installer.rb:20:in `install_template'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/template/processor/template_declaration.rb:32:in `install'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/template/processor/template_processor.rb:42:in `block in install_templates'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/template/processor/template_processor.rb:40:in `each'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/template/processor/template_processor.rb:40:in `install_templates'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/lib/generamba/cli/template/template_install_command.rb:18:in `install'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/invocation.rb:115:in `invoke'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor.rb:235:in `block in subcommand'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
     lib/ruby/gems/2.3.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
     lib/ruby/gems/2.3.0/gems/generamba-1.2.1/bin/generamba:5:in `<top (required)>'
     bin/generamba:23:in `load'
     bin/generamba:23:in `<main>'
```

**Comment by Antowkos** [comment]
@Beniamiiin @etolstoy I don't want to upset you, but there are also one more issue related to this feature. When i try to call `generamba template install` with template which is using this new feature i got same error as described above. 

BTW template is working fine if i just use it by copy-paste to templates folder.

**Comment by etolstoy** [comment]
`generamba template install` fixed in 1.2.2, thanks everyone.
